src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ServiceUtilHelper.cs

```diff
@@ -199,6 +216,80 @@ public static void EnsureClientCertificateInstalled()
         ThrowIfClientCertificateInstallationError();
     }
 
+    // Tries to ensure that the server machine certificate is installed into
+    // the TrustedPeople store.  It obtains the server certificate from the service
+    // utility endpoint and either installs it or verifies that a matching
+    // one is already installed.  InvalidOperationException will be thrown
+    // if an error occurred attempting to install the certificate.  This
+    // method may be called multiple times but will attempt the installation
+    // once only.
+    public static void EnsureServerCertificateInstalled()
```

Comment:
I almost wonder if we should be using an entirely different certificate here rather than rely on the existing server cert. It's not going to be immediately apparent whether this works because the certificate is chain building properly, or whether the certs are in the trusted people store. 

Ideally, shouldn't the server present a certificate that _can't_ chain build to our Root CA cert? Then it would be obvious and provable that we used an otherwise-non-valid cert to authenticate to a server that would otherwise not be trusted